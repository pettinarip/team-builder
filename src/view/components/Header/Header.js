import React from 'react'
import PropTypes from 'prop-types'

const Header = ({ authenticated, signOut }) => (
  <header>
    <h1 className='super-title'>Team lineup</h1>
    <p>Build your team and shear it with your teammates!</p>
    <div>
      { authenticated && <button type='button' onClick={signOut}>Sign out</button> }
    </div>
  </header>
)

Header.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  signOut: PropTypes.func.isRequired
}

export default Header
