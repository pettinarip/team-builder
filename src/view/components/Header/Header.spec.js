import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'

describe('Header component', () => {
  it('should render the component properly', () => {
    const component = mountHeader()
    expect(component).toMatchSnapshot()
  })
})

function mountHeader (props) {
  const propsToUse = {
    signOut () {},
    authenticated: false,
    ...props
  }

  return shallow(<Header {...propsToUse} />)
}
