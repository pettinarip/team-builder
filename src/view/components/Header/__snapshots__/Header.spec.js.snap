// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render the component properly 1`] = `
<header>
  <h1
    className="super-title"
  >
    Team Builder
  </h1>
  <p>
    Build your team and shear it with your teammates!
  </p>
  <div />
</header>
`;

exports[`Header component should render the sign out button if it is authenticated 1`] = `
<header>
  <h1
    className="super-title"
  >
    Team Builder
  </h1>
  <p>
    Build your team and shear it with your teammates!
  </p>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Sign out
    </button>
  </div>
</header>
`;
